### PR TITLE
Run style tests with all properties included

### PR DESCRIPTION
**Pull request by Manishearth**
This turns on all properties (including stylo-only ones) when the style unit tests are compiled. This lets us test the parsing of gecko-only properties.

These tests can't go in tests/stylo since this introduces dependencies on `Gecko_*` symbols. This method lets us test parsing and serialization of geckolib properties compiled in Servo mode (e.g. using Servo atoms, Servo URLs)

r? @emilio

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13386)

### CODE REVIEW

File: components/style/build.rs [hunk 1]
```diff
@@ -55,6 +55,7 @@ fn main() {
         .arg(&script)
         .arg(product)
         .arg("style-crate")
+        .arg(if cfg!(feature = "testing") {"testing"} else {"regular"})
```
Comment (emilio) [inline components/style/build.rs]: I'm surprised tidy doesn't complain about the spacing around braces here

File: components/style/properties/build.py [hunk 2]
```diff
@@ -23,10 +23,12 @@ def main():
         abort(usage)
     product = sys.argv[1]
     output = sys.argv[2]
+    testing = len(sys.argv) > 3 and sys.argv[3] == "testing"
```
Comment (emilio) [inline components/style/properties/build.py]: We're always passing this argument now, right? can you make this just `sys.argv[3] == "testing"`?

**Review by emilio — COMMENTED** [review]
r=me, ideally with a doc comment in the data file about the testing stuff and why we do it this way.

Thanks!

File: components/style/properties/build.py (hunk 2 above)
Comment (Manishearth) [inline components/style/properties/build.py]: We're not always passing this argument; style_tests itself contains a test that calls this without the third argument. I could add it, but it seems better to ignore.

Comment (Manishearth) [inline components/style/properties/build.py]: Eh, I'll add it

**Comment by Manishearth** [comment]
@bors-servo retry

Looks like the bug mentioned in IRC earlier, but can't be related with this. Let's see if it's an intermittent.

File: components/style/properties/build.py [hunk 3]
```diff
@@ -18,15 +18,17 @@
 
 
 def main():
-    usage = "Usage: %s [ servo | gecko ] [ style-crate | html ]" % sys.argv[0]
+    usage = "Usage: %s [ servo | gecko ] [ style-crate | html ] [ testing | regular ]" % sys.argv[0]
     if len(sys.argv) < 3:
```
Comment (emilio) [inline components/style/properties/build.py]: nit: This needs to become `< 4`, right?

**Comment by Manishearth** [comment]
@bors-servo r=emilio

done

**Pull request merged**